### PR TITLE
Add metric when history replication message is too large

### PR DESCRIPTION
**Pull request by bowenxia**
**What changed?**
Add metric when history replication message is too large

**Why?**
to catch a previous on-call issue

**How did you test it?**

**Potential risks**

**Release notes**

**Documentation Changes**

### CODE REVIEW

File: service/history/handler/handler.go [hunk 1]
```diff
@@ -1601,6 +1601,9 @@ func (h *handlerImpl) GetReplicationMessages(
 
 		size := proto.FromReplicationMessages(tasks).Size()
 		if (responseSize + size) >= maxResponseSize {
+			metricsScope := h.GetMetricsClient().Scope(metrics.ReplicationMessageTooLargeScope, metrics.ShardIDTag(int(shardID)))
```
Comment (Shaddoll) [inline service/history/handler/handler.go:1604]: let's reuse the metric scope returned in line 1558

Comment (bowenxia) [reply service/history/handler/handler.go]: changed. I only added the shard tag inside of that if statement.

File: service/history/handler/handler.go [hunk 2]
```diff
@@ -1601,6 +1601,9 @@ func (h *handlerImpl) GetReplicationMessages(
 
 		size := proto.FromReplicationMessages(tasks).Size()
 		if (responseSize + size) >= maxResponseSize {
+			metricsScope.Tagged(metrics.ShardIDTag(int(shardID)))
```
Comment (taylanisikdemir) [inline service/history/handler/handler.go:1604]: it would be better to have a gauge metric instead of counter which would have value `responseSize + size`

**Review by Shaddoll — APPROVED** [review]

File: service/history/handler/handler.go (hunk 2 above)
Comment (taylanisikdemir) [reply service/history/handler/handler.go]: I see. If we are more interested in the number of occurrences then counter makes sense.

**Review by taylanisikdemir — APPROVED** [review]

**Pull request merged**